docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,196 @@
+we should retain all data produced by EmailAlertApi with the exception of
+emails which are [archived] to S3 & [deleted] after a week.
+
+Since then we've come to realise that a revision of our data retention policy
+is needed as we're generating more data that will never be used again - by us,
+or the system.
+
+This was initially prompted by recent work to [migrate subscribers to daily
+digest emails] which lead to an increase in the amount of unmanaged data we
+produce and store locally in EmailAlertApi. Consequently, this provoked worries
+that perpetually keeping data could result in capacity issues due to the rapid
+increase in data being stored. An example being we now generate ~430,000
+`DigestRunSubscriber` records each day which is about eight times the amount
+compared to that of ten months ago.
+
+We've also gained more insight into what data we're not using for analytics.
+The original plan documented in the previous retention policy was to retain as
+much data as possible to enable us to do as much analysis as needed in the
+future. However, much of this data is never used for analysis and so doesn't
+serve a purpose in the email system.
+
+Contrary to what was previously thought, storing lots of unused data
+indefinitely actually complicates and pollutes our auditing and analysis. An
+example of this can be seen if we look at subscriber lists. We have over
+~73,000 lists with active subscribers but in the last week alone, we've
+accumulated 1139 lists which don't have any subscriptions at all (ended or
+active). These unused lists prevent us from doing simple auditing tasks
+easily, like finding which lists to report on.
+
+Knowing the above, we thought about how we could cleanup this unused data to
+keep EmailAlertApi's database manageable going forward. There are a couple of
+different approaches to consider, including storing unused data in cloud
+storage away from the system's database (like we do when archiving emails) and
+subsequently deleting the data locally. Or, setting up a means of removing the
+data we don't use from the system entirely.
+
+[archived]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[deleted]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_deletion_worker.rb
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+## Decision
+
+We decided that historic, unused data should be removed periodically from the
+system entirely. It was concluded that the data the system no longer uses
+should only exist for a maximum of 1 year before removal. We sometimes
+analyse historic data from up to a year ago but we have no evidence of needing
+this data for analytics beyond a year.
+
+Emails (and their associated SubscriptionContents) are the one place where we
```

Comment:
Minor: this paragraph could go after the following list, which relates to the paragraph above.